gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -137,11 +142,15 @@ public GobblinTaskRunner(String applicationName, String helixInstanceName, Strin
       String taskRunnerId, Config config, Optional<Path> appWorkDirOptional)
       throws Exception {
     this.helixInstanceName = helixInstanceName;
-    this.config = config;
     this.taskRunnerId = taskRunnerId;
 
     Configuration conf = HadoopUtils.newConfiguration();
-    this.fs = buildFileSystem(this.config, conf);
+    this.fs = buildFileSystem(config, conf);
+    Path appWorkDir = appWorkDirOptional.isPresent() ? appWorkDirOptional.get()
+        : GobblinClusterUtils
+            .getAppWorkDirPathFromConfig(config, this.fs, applicationName, applicationId);
+
+    this.config = saveConfigToFile(config, appWorkDir);
```

Comment:
we should only need this if we are running task in separate process. so should this be within a check?

Reply:
Correct.
However, there is no harm in generating this config. In fact, it is useful as a debugging aid.

Reply:
Also seems like all tasks will save to the same location? unless I am missing something.

Reply:
No. The config is cluster level config. It will only be saved once per worker process when it starts up.

Reply:
The child processes will read from the same location. 
Also, the assumption is that if there are multiple worker processes running on the same host, they will start from different work directory.

Reply:
Thanks, just saw that this gets initialized once. Its a fair assumption that each worker process will have its own work directory.